records/generator_test.go

```diff
@@ -244,8 +262,16 @@ func TestInsertState(t *testing.T) {
 		{rgDocker.As, "nginx.marathon.mesos.", []string{"1.2.3.11"}},
 		{rgDocker.As, "car-store.marathon.slave.mesos.", []string{"1.2.3.11"}},
 	} {
-		if got := tt.rrs[tt.name]; !reflect.DeepEqual(got, tt.want) {
-			t.Errorf("test #%d: %q: got: %q, want: %q", i, tt.name, got, tt.want)
+		// convert want into a map[string]struct{} for simpler comparison
+		want := map[string]struct{}{}
```

Comment:
Specifics:
-Why are we converting the datastructure?
-What are we actually testing for?